bin/pycbc_optimize_snr

```diff
@@ -186,12 +203,22 @@ bounds = [(minchirp, maxchirp), (0.01, 0.2499), (-0.9, 0.9), (-0.9, 0.9)]
 
 logging.info('Starting optimization')
 
-results = differential_evolution(compute_network_snr, bounds,
-                                 maxiter=100, workers=(args.cores or -1),
-                                 popsize=200, mutation=(0.5,1),
+try:
+
+    if args.optimizer == 'differential_evolution':
+        results = differential_evolution(compute_network_snr, bounds,
+                                 maxiter=args.maxiter, workers=(args.cores or -1),
+                                 popsize=args.popsize, mutation=(0.5,1),
                                  recombination=0.7, callback=callback_func,
                                  args=extra_args)
 
+    elif args.optimizer == 'shgo':
+        results = shgo(compute_network_snr, bounds=bounds, args=extra_args, 
+                iters=args.iters, n=args.n, sampling_method="sobol")
+               
+except RuntimeError: 
+    print("You haven't specified an optimizer, either shgo or differential_evolution")
+
```

Comment:
I am not sure a try/except statement is the way to go here, or that it will even do what you want. I recommend using the `choices` argument of `parser.add_argument()` instead, which will ensure that `args.optimizer` will have a valid value.